canvas_sdk/v1/data/questionnaire.py

```diff
@@ -0,0 +1,165 @@
+from canvas_sdk.v1.data.base import CommittableModelManager
+from canvas_sdk.v1.data.user import CanvasUser
+
+
+class ResponseOptionSet(models.Model):
+    """ResponseOptionSet."""
+
+    class Meta:
+        managed = False
+        app_label = "canvas_sdk"
+        db_table = "canvas_sdk_data_api_responseoptionset_001"
+
+    dbid = models.BigIntegerField(primary_key=True)
+    created = models.DateTimeField()
+    modified = models.DateTimeField()
+    status = models.CharField()
+    name = models.CharField()
+    code_system = models.CharField()
+    code = models.CharField()
+    type = models.CharField()
+    use_in_shx = models.BooleanField()
+
+
+class ResponseOption(models.Model):
+    """ResponseOption."""
+
+    class Meta:
+        managed = False
+        app_label = "canvas_sdk"
+        db_table = "canvas_sdk_data_api_responseoption_001"
+
+    dbid = models.BigIntegerField(primary_key=True)
+    created = models.DateTimeField()
+    modified = models.DateTimeField()
+    status = models.CharField()
+    name = models.CharField()
+    code = models.CharField()
+    code_description = models.CharField()
+    value = models.CharField()
+    response_option_set = models.ForeignKey(
+        ResponseOptionSet, on_delete=models.CASCADE, related_name="options"
+    )
+    ordering = models.IntegerField()
+
+
+class Question(models.Model):
+    """Question."""
+
+    class Meta:
+        managed = False
+        app_label = "canvas_sdk"
+        db_table = "canvas_sdk_data_api_question_001"
+
+    id = models.UUIDField()
+    dbid = models.BigIntegerField(primary_key=True)
+    created = models.DateTimeField()
+    modified = models.DateTimeField()
+    status = models.CharField()
+    name = models.CharField()
+    response_option_set = models.ForeignKey(
+        ResponseOptionSet, on_delete=models.CASCADE, related_name="questions"
+    )
+    acknowledge_only = models.BooleanField()
+    show_prologue = models.BooleanField()
+    code_system = models.CharField()
+    code = models.CharField()
+
+
+class Questionnaire(models.Model):
+    """Questionnaire."""
+
+    class Meta:
+        managed = False
+        app_label = "canvas_sdk"
+        db_table = "canvas_sdk_data_api_questionnaire_001"
+
+    id = models.UUIDField()
+    dbid = models.BigIntegerField(primary_key=True)
+    created = models.DateTimeField()
+    modified = models.DateTimeField()
+    status = models.CharField()
+    # graph_header
+    # graph_nodes
+    # graph_edges
+    name = models.CharField()
+    expected_completion_time = models.FloatField()
+    can_originate_in_charting = models.BooleanField()
+    use_case_in_charting = models.CharField()
+    scoring_function_name = models.TextField()
+    scoring_code_system = models.CharField()
+    scoring_code = models.CharField()
+    code_system = models.CharField()
+    code = models.CharField()
+    search_tags = models.CharField()
+    # TODO: Need to bring over questionnaire question map?
```

Comment:
Yeah, I think so. The idea, which I don't think has been realized, was that we could create well coded questions with well coded responses and compose them across several questionnaires. Reusable questions, essentially. I don't think we've ever done that, and all questions exist in exactly one questionnaire.

Reply:
The `questionnairequestionmap` table does the many to many association of questions to questionnaires, so we do already have reusable questions (unless you mean something different)

Reply:
Yes, the data model allows for question reuse, but I mean they aren't reused. I don't believe anyone takes advantage of that capability.